From 8b3632f2f0d7f061a30174218f82eaa199cf9f43 Mon Sep 17 00:00:00 2001
From: Andrew Pinski <Andrew.Pinski@caviumnetworks.com>
Date: Sun, 31 Jan 2021 19:50:06 +0530
Subject: [PATCH] arm64: ensure the kernel is compiled for LP64 The kernel
 needs to be compiled as a LP64 binary for ARM64, even when using a compiler
 that defaults to code-generation for the ILP32 ABI. Consequently, we need to
 explicitly pass '-mabi=lp64' (supported on gcc-4.9 and newer).

Signed-off-by: Andrew Pinski <Andrew.Pinski@caviumnetworks.com>
Signed-off-by: Philipp Tomsich <philipp.tomsich@theobroma-systems.com>
Signed-off-by: Christoph Muellner <christoph.muellner@theobroma-systems.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
Reviewed-by: David Daney <ddaney@caviumnetworks.com>
Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Shashank Baghel <theradcolor@gmail.com>
---
 arch/arm64/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index c20987a87141..a9a45e74a180 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -86,14 +86,19 @@ else
 KBUILD_CFLAGS	+= $(call cc-ifversion, -ge, 0500, -DCONFIG_ARCH_SUPPORTS_INT128)
 endif
 
+KBUILD_CFLAGS	+= $(call cc-option,-mabi=lp64)
+KBUILD_AFLAGS	+= $(call cc-option,-mabi=lp64)
+
 ifeq ($(CONFIG_CPU_BIG_ENDIAN), y)
 KBUILD_CPPFLAGS	+= -mbig-endian
 AS		+= -EB
 LD		+= -EB
+LDFLAGS		+= -maarch64linuxb
 else
 KBUILD_CPPFLAGS	+= -mlittle-endian
 AS		+= -EL
 LD		+= -EL
+LDFLAGS		+= -maarch64linux
 endif
 
 CHECKFLAGS	+= -D__aarch64__
-- 
2.30.0

